frontend/src/components/software/SoftwareDetailsContainer.jsx

```diff
@@ -2,9 +2,60 @@ import React, {Component} from 'react';
 import SoftwareDetailsComponent from './SoftwareDetailsComponent';
 
 class SoftwareDetailsContainer extends Component {
+  constructor(props) {
+    super(props);
+    this.addCall = this.addCall.bind(this);
+    this.state = {
+      name: '',
+      key: '',
+      availableKeys: '',
+      duration: '',
+      mode: '',
+      id: '',
+    };
+  }
+
+
+
+  addCall(){
```

Comment:
Ciekawostka: jeśli tutaj użyjemy konstrukcji `addCall = () => {`, to takiej funkcji nie trzeba wtedy bindować, jak masz powyżej w konstruktorze 😎

Reply:
poprawione